src/codegen/compile.ml

```diff
@@ -3892,6 +3893,18 @@ module IC = struct
       edesc = nr (FuncExport (nr fi))
     })
 
+  let export_timer env =
+    assert (E.mode env = Flags.ICMode || E.mode env = Flags.RefMode);
+    let fi = E.add_fun env "canister_global_timer"
+      (Func.of_body env [] [] (fun env ->
+        G.i (Call (nr (E.built_in env "timer_exp"))) ^^
+        GC.collect_garbage env))
```

Comment:
really?

Reply:
We probably need it. It could be made cheaper if we'd compare the HP before with after and only call `GC.collect_garbage` when there is a change. Similarly for `heartbeat`.

Reply:
I wouldn't special case anything. Just let the gc do it's thing.